### DISCUSSION TITLE
OS Platform Share Dialog Support?

### DISCUSSION THREAD

**Discussion by scottkuhl** [AvaloniaUI/Avalonia#15087] [Q&A]
Does Avalonia have anything already built in to support platform level sharing and targeting?

I am looking for something similar to this:  https://learn.microsoft.com/en-us/dotnet/maui/platform-integration/data/share?view=net-maui-8.0

![image](https://github.com/AvaloniaUI/Avalonia/assets/4465616/445e97ce-0d33-4bac-adf5-d559b014dd31)

**Comment by thevortexcloud** [comment]
You can either use MAUI directly, or Xamarin. Either works with Avalonia.

Here is the Xamarin docs:
https://learn.microsoft.com/en-us/xamarin/essentials/share?tabs=android

And here is the docs on how to use MAUI with Avalonia:
https://github.com/AvaloniaUI/AvaloniaMauiHybrid

Avalonia is primarily a UI framework. The vast majority of the platform specific integrations (especially on mobile) is handled by other frameworks such as Xamarin.

**Comment by scottkuhl** [comment]
Xamarin and Maui only seem to support sharing, not acting as a share target.  Is there any way to act as the target, or do you need to write platform specific code?

**Reply by thevortexcloud** [reply]
As far as I am aware (I am not a mobile/Xamarin expert so I may just be missing something) you will need to resort to platform specific APIs for that. Xamarin looks like it already has the bindings for it on Android.

https://learn.microsoft.com/en-us/samples/xamarin/monodroid-samples/google-services-nearby-connectionsquickstart/

However like most useful android features, it looks like Google is possibly deprecating it and replacing it with a closed source app.

https://blog.google/products/android/ces-2024-android-updates/#:~:text=A%20new%20way,devices%20next%20month

Some google searches seem to suggest Xamarin can do air drop iOS too.
